### PR TITLE
[Neuropixels] Add time shift to processed behavior data timestamp columns

### PR DESCRIPTION
**Pull request by weiglszonja**
To align with Bpod, this PR implements the shifting of timestamp columns in the processed behavior data.
The columns that need to be shifted:
- "Cled"
- "Lled"
- "Rled"
- "l_opt"
- "r_opt"

See #31 

This PR also includes the changes how we are storing "Lled", "Rled","l_opt" and "r_opt" columns.
Before we added the 4 timestamp for each trial in the same column, just as:
Screenshot 2024-12-02 at 18 36 15 (2)

The 4 timestamp for each trial is from:

e.g. Lled = [left reward side light on, left  reward side light off,  left reward side light flash on, left reward side light flash off] 

l_opt = [left side light turns on, left side light turns off, right port turns off,  side light turns off]

> During the delay between the center light turning off and the reward arriving, the side light turns on. The side light turns off when the reward is available, then stays off until the animal collects the reward. When the animal nose pokes to collect the reward, the light flashes on/off.
Alternatively, if the animal opts out, there is a flash on the opt out port on/off. Then the reward port turns off. 

I think it is better not to store them as arrays in the trials table but break them into separate columns, e.g.

rewarded_port_onset_time -> corresponds to the first timestamp of "Lled" if the rewarded side for that trials was the left port
rewarded_port_offset_time -> corresponds to the second timestamp of "Lled" if the rewarded side for that trials was the left port
rewarded_port_flash_onset_time -> corresponds to the third timestamp of "Lled" if the rewarded side for that trials was the left port
rewarded_port_flash_offset_time -> corresponds to the fourth timestamp of "Lled" if the rewarded side for that trials was the left port

We populate rewarded_port_onset_time, rewarded_port_offset_time, rewarded_port_flash_onset_time, rewarded_port_flash_offset_time etc. for each trial depending on which side was the rewarded side.

### CODE REVIEW

**Comment by weiglszonja** [comment]
[Example file](https://us-east-2.console.aws.amazon.com/s3/object/catalystneuro-processing?region=us-east-2&bucketType=general&prefix=constantinople-lab-to-nwb/J076_2023-12-12_14-52-04.nwb)

Screenshot of the trials table:
Screenshot 2024-12-02 at 18 45 09 (2)

**Review by alessandratrapani — APPROVED** [review]
Just a minor comment, the rest looks good

File: src/constantinople_lab_to_nwb/schierek_embargo_2024/schierek_embargo_2024_notes.md [hunk 1]
```diff
@@ -129,27 +129,77 @@ column_descriptions = dict(
...
+>>> [48.57017236918037, 68.2978722016674, 72.2138230625031, 75.45578122765313, 87.98392024937102, 106.3281781420765, 108.04842315623304]
+```
+
+## Alignment
+
+We are aligning the starting time of the recording and sorting interfaces to the Bpod interface.
+
+We are computing the time shift from the Bpod trial start time to the NIDAQ trial start time.
+
+```python
```
Comment (weiglszonja) [inline src/constantinople_lab_to_nwb/schierek_embargo_2024/schierek_embargo_2024_notes.md:174]: From a conceptual point of view, the trial start and the central port onset are equivalent: a trial starts when the first time the central port is on.

File: src/constantinople_lab_to_nwb/schierek_embargo_2024/schierek_embargo_2024_notes.md [hunk 2]
```diff
@@ -129,27 +129,77 @@ column_descriptions = dict(
...
+
+center_port_onset_times[:7]
+>>> [48.57017236918037, 68.2978722016674, 72.2138230625031, 75.45578122765313, 87.98392024937102, 106.3281781420765, 108.04842315623304]
+```
+
+## Alignment
+
+We are aligning the starting time of the recording and sorting interfaces to the Bpod interface.
+
+We are computing the time shift from the Bpod trial start time to the NIDAQ trial start time.
```
Comment (weiglszonja) [inline src/constantinople_lab_to_nwb/schierek_embargo_2024/schierek_embargo_2024_notes.md:172]:
```suggestion
We are computing the time shift from the Bpod trial start time to the NIDAQ trial start time.

From a conceptual point of view, the trial start and the central port onset are equivalent: a trial starts when the first time the central port is on.

```

File: src/constantinople_lab_to_nwb/schierek_embargo_2024/schierek_embargo_2024_notes.md [hunk 3]
```diff
@@ -129,27 +129,77 @@ column_descriptions = dict(
...
+
+```python
+time_shift = bpod_trial_start_times[0] - center_port_onset_times[0]
+>>> -28.58217236918037
+```
```
[the remark this answers is not in the document]
Comment (weiglszonja) [reply src/constantinople_lab_to_nwb/schierek_embargo_2024/schierek_embargo_2024_notes.md:180]: Thanks, added!

**Pull request merged**